docs/proposal/kmesh_support_localityLB.md

```diff
@@ -0,0 +1,127 @@
+
+What is locality strict mode? In locality strict mode, the LB (load balancing) algorithm will only select backends that exactly match the routingPreference. This means that in such a mode, the load balancer enforces a strict policy where it only routes traffic to backends based on a perfect match with the specified locality preferences, ensuring that requests are handled by servers that meet specific criteria related to their location or other attributes.
+
+### Design Details
+
+Introduce a priority (rank) between the existing service and endpoint to achieve finer-grained traffic control and service management. The newly designed endPointKey will consist of the service ID, priority level, and backend index, with its corresponding value being the backendKey. Development will be carried out in both user space and kernel space to implement the functionality.
+
+![locality_lb_pic](pics/locality_lb.svg)
+
+#### Control plane（user-space）
+1. The user-space kmesh process needs to maintain locality, clusterId, and network information, which is updated when the xDS receives information about the workload currently on that node.
+2. The user-space kmesh process also needs to maintain routingPreference, which is updated when the xDS receives information about the service currently on that node.
+3. Once the above information has been successfully obtained by the user-space, for each incoming workload, its locality information is compared with the kmesh process's locality to calculate a priority.
+   Calculation Process: Match according to the routing rules defined in the service load balancing scope. For example, if a service is configured with a region, the kmesh process's region is compared with the target backend's region. If they are the same, the priority value is incremented. After calculating the rank, the corresponding endpointKey is generated using the service ID, backend ID, and rank, thereby associating the service with the backend.
+
+#### Data plane（kernel-space）
+When a request is made to a service, the BPF program traverses the endpoint map to query the backend ID based on the requested service ID and count, following the priority rank. 
+- In strict mode, only backends that match all scopes are considered.
+- In failover mode, the BPF program will iterate through the ranks starting from the highest and moving downward. If there are backends stored at a particular priority level, one of the queried backends is selected at random to serve the request.
+
+#### data struct
+1. workload.h
+```
+typedef struct {
+    __u32 service_id;    // service id
+    __u32 prio;          // prio means rank, 6 means match all, and 0 means match nothing
+    __u32 backend_index; // if endpoint_count = 3, then backend_index = 0/1/2
+} endpoint_key;
+```
+
+2. service
+
+workload.h
+```
+#define MAX_PRIO                  6
```

Comment:
IIUC the max priority  rank can be `7`

 with 
	// Prefer traffic in the same region.
	LoadBalancing_REGION LoadBalancing_Scope = 1
	// Prefer traffic in the same zone.
	LoadBalancing_ZONE LoadBalancing_Scope = 2
	// Prefer traffic in the same subzone.
	LoadBalancing_SUBZONE LoadBalancing_Scope = 3
	// Prefer traffic on the same node.
	LoadBalancing_NODE LoadBalancing_Scope = 4
	// Prefer traffic in the same cluster.
	LoadBalancing_CLUSTER LoadBalancing_Scope = 5
	// Prefer traffic in the same network.
	LoadBalancing_NETWORK LoadBalancing_Scope = 6

Reply:
prio value is from 0 to 6, so i set MAX_PRIO as 6, which actually is the 7th rank.